### PR TITLE
Use `libctr/cgroups` for stats collection

### PR DESCRIPTION
**Pull request by Mo-Fatah**
#### What type of PR is this?

/kind dependency-change

#### What this PR does / why we need it:
Use `libcontainer/cgroups` to collect sandbox/container stats instead of `containers/common`. This will be useful once we move on from cAdvisor. 
#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: internal/config/cgmgr/cgroupfs_linux.go
Comment (haircommander) [inline internal/config/cgmgr/cgroupfs_linux.go:103]: so this will create a new entry in the map, you need to do `cgMgr, ok := ..; ok && cgMgr != nil {` to not add an entry

Comment (haircommander) [inline internal/config/cgmgr/cgroupfs_linux.go:103]: I think we need some locking here

File: internal/config/cgmgr/cgroupfs_linux.go
Comment (haircommander) [inline internal/config/cgmgr/cgroupfs_linux.go:101]: Naming suggestion: golang sugggests not prepending `Get` to getters. Also ,the other functions in this package spell out Container, so maybe `ContainerCgroupManager` ?

File: internal/config/cgmgr/cgroupfs_linux.go
Comment (haircommander) [reply internal/config/cgmgr/cgroupfs_linux.go:103]: oh shoot sorry I commented this on the wrong PR, lemme move this convo over there

File: internal/config/cgmgr/cgmgr_linux.go
Comment (haircommander) [inline internal/config/cgmgr/cgmgr_linux.go:64]: is this called anywhere? I think I'd prefer to not have this endpoint for now. The caller of ContainerCgroupManager can implement it, no?

File: internal/config/cgmgr/cgroupfs_linux.go
Comment (haircommander) [inline internal/config/cgmgr/cgroupfs_linux.go:124]: couldn't we just directly access the map? doesn't GetCtr... create the cgroup manager if it doesn't exist?

**Comment by haircommander** [comment]
handful of comments, also before this merges the commits should be squashed

File: internal/config/cgmgr/cgroupfs_linux.go
Comment (Mo-Fatah) [reply internal/config/cgmgr/cgroupfs_linux.go:124]: To destroy the cgroup, we should get it from the map (if we use cgroup v1), or create the cgroup object on the fly for cgroup v2. `GetCtrCgroupManager` internally does this.

Comment (haircommander) [reply internal/config/cgmgr/cgroupfs_linux.go:124]: FYI I don't think we should destroy the cgruop in these functions, that's done by the runtime when the container is cleaned up. We should just remove the entry from the map, if present

File: internal/config/cgmgr/cgmgr_linux.go
Comment (Mo-Fatah) [reply internal/config/cgmgr/cgmgr_linux.go:64]: it isn't called anywhere yet, not in this PR. Later I am planning of removing `PopulateContainerCgroupStats` and similar functions, and use `ContainerCgroupStats` and `SandboxCgroupStats`. Both  `ContainerCgroupStats` and `SandboxCgroupStats` should be called directly from the stats server's loop.

**Comment by Mo-Fatah** [comment]
@haircommander can you have another look now ?

File: internal/config/cgmgr/cgmgr_linux.go
Comment (haircommander) [inline internal/config/cgmgr/cgmgr_linux.go:66]: I think we should keep the names consistent. RemoveContainer* and RemoveSandbox* below

**Review by saschagrunert — APPROVED** [review]

**Comment by saschagrunert** [comment]
/hold 

for @haircommander final look

**Comment by haircommander** [comment]
/hold cancel

well done, thank you!

**Pull request merged**
